src/analyses/ai.cpp

```diff
@@ -248,38 +248,28 @@ bool ai_baset::visit(
   bool new_data=false;
 
   // Function call and end are special cases
-  if(l->is_function_call() && !goto_functions.function_map.empty())
+  if(l->is_function_call())
   {
     DATA_INVARIANT(
       goto_program.get_successors(l).size() == 1,
       "function calls only have one successor");
+
+    locationt to_l = std::next(l);
     DATA_INVARIANT(
-      *(goto_program.get_successors(l).begin()) == std::next(l),
+      *(goto_program.get_successors(l).begin()) == to_l,
```

Comment:
Inline maybe? Seems we never use `to_l` again.

Reply:
Inlined.